### PR TITLE
Track call support for RETL

### PR DESCRIPTION
**Pull request by ishachopra0401**
## What do these changes do?

Added docs for Track call support for RETL

Preview URL: https://rudderstack-docs-git-retl-track-call-rudder-stack.vercel.app/sources/reverse-etl/features/syncing-events/

### CODE REVIEW

**Review by vyeshwanth — APPROVED** [review]
LGTM!

File: docs/sources/reverse-etl/features/syncing-events.mdx [hunk 1]
```diff
@@ -0,0 +1,40 @@
+---
+title: "Syncing Events"
+description: >-
+ Detailed technical description on syncing event data from Reverse ETL sources to the destinations.
+---
+
+RudderStack supports syncing event data from Reverse ETL sources to the downstream destinations using the <Link to="/event-spec/standard-events/identify/">`identify`</Link> and <Link to="/event-spec/standard-events/track/">`track`</Link> calls.
```
Comment (ameypv-rudder) [inline docs/sources/reverse-etl/features/syncing-events.mdx:7]:
```suggestion
RudderStack supports syncing data from the <Link to="/sources/reverse-etl/#supported-reverse-etl-sources">supported  Reverse ETL sources</Link> to the downstream destinations as <Link to="/event-spec/standard-events/identify/">`identify`</Link> and <Link to="/event-spec/standard-events/track/">`track`</Link> calls.
```

File: docs/sources/reverse-etl/features/syncing-events.mdx [hunk 2]
```diff
@@ -0,0 +1,40 @@
+---
+title: "Syncing Events"
+description: >-
+ Detailed technical description on syncing event data from Reverse ETL sources to the destinations.
+---
```
Comment (ameypv-rudder) [inline docs/sources/reverse-etl/features/syncing-events.mdx:5]:
```suggestion
aliases: ["Reverse ETL syncs"]
---
```

File: docs/sources/reverse-etl/features/syncing-events.mdx (hunk 1 above)
Comment (ameypv-rudder) [reply docs/sources/reverse-etl/features/syncing-events.mdx:7]: The data residing in the warehouse can be any data, not just event data. RS ingests this data and sends it to the downstream tools as either `identify` or `track` events. Please correct me if my understanding is wrong here.

File: docs/sources/reverse-etl/features/syncing-events.mdx [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+RudderStack supports syncing event data from Reverse ETL sources to the downstream destinations using the <Link to="/event-spec/standard-events/identify/">`identify`</Link> and <Link to="/event-spec/standard-events/track/">`track`</Link> calls.
+
+<div class="warningBlock">
+You can use the <code class="inline-code">track</code> call only while mapping your warehouse columns using the JSON builder and not the <Link to="/sources/reverse-etl/features/visual-data-mapper/">Visual Data Mapper</Link>.
```
Comment (ameypv-rudder) [inline docs/sources/reverse-etl/features/syncing-events.mdx:10]: Should we say something like "You can send your data as <code class="inline-code">track</code> calls only while mapping your warehouse columns to the destination fields using the <Link to="/sources/reverse-etl/common-settings/importing-data-using-tables/">JSON builder</Link>. <Link to="/sources/reverse-etl/features/visual-data-mapper/">Visual Data Mapper</Link> is not supported currently.

File: docs/sources/reverse-etl/features/syncing-events.mdx [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+</div>
+
+## Key features
+
+Some key features of syncing events from Reverse ETL sources are listed below:
+
+- **Ensure the reliability of your events**: Sometimes, your critical events (such as purchases) may get blocked by things outside your control, such as Adblock. In such cases, if you have all your users' activity in the database, you can send it to the downstream systems as a guaranteed source of truth.
```
Comment (ameypv-rudder) [inline docs/sources/reverse-etl/features/syncing-events.mdx:17]:
```suggestion
- **Ensure the reliability of your events**: Sometimes, your critical events(such as purchases) may get blocked by things outside your control, such as adblockers. In such cases, if you have all your users' activity in the database, you can send it to the downstream systems as a guaranteed source of truth.
```

File: docs/sources/reverse-etl/features/syncing-events.mdx [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+- **Simulate historical tracking of events**: You can replay the historical data from other data sources to your sales, support, and marketing systems to maintain business continuity.
+- **Fix bad data and replay it**: There is a possibility that your destination ends up with some bad data due to a bug in the tracking code or any other reason.
+In such cases, you can clear the bad data and replay a clean version by syncing the events.
+- **Test your transformations and integrations**: You can use a sample dataset to test an event stream for a new version of your transformation or a new integration using the `track` call. 
+
+## Use cases for sending `track` events
+
+Some examples where you can use the `track` call to send events from your warehouse into the downstream destinations are listed below:
+
+- Sending event data to engagement platforms such as Braze, Iterable, CustomerIO, or Salesforce Marketing Cloud to trigger behavior or populate user actions.
```
Comment (ameypv-rudder) [inline docs/sources/reverse-etl/features/syncing-events.mdx:27]:
```suggestion
- Sending event data to engagement platforms such as Braze, Iterable, Customer.io, or Salesforce Marketing Cloud to trigger behavior or populate user actions.
```

File: docs/sources/reverse-etl/features/syncing-events.mdx [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+- Sending event data to the downstream systems that are configured to handle `track` calls like Kafka or Kinesis.
```
Comment (ameypv-rudder) [inline docs/sources/reverse-etl/features/syncing-events.mdx:28]:
```suggestion
- Sending event data to the downstream systems that are configured to handle `track` calls like Apache Kafka or Amazon Kinesis.
```

File: docs/sources/reverse-etl/features/syncing-events.mdx [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+- Replaying the fact table that might be the source of truth for user behavior to the downstream tools such as StatSig, Optimizely, etc.
```
Comment (ameypv-rudder) [inline docs/sources/reverse-etl/features/syncing-events.mdx:29]:
```suggestion
- Replaying the fact table that might be the source of truth for user behavior to the downstream tools such as Statsig, Optimizely, etc.
```

File: docs/sources/reverse-etl/features/syncing-events.mdx [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+
+## Steps for syncing event data
+
+To sync the event data from a Reverse ETL source to a downstream destination, follow these steps:
+
+1. Configure your Reverse ETL source using the <Link to="/sources/reverse-etl/">source-specific settings</Link>.
+2. Connect the source to the required destination by configuring the destination settings as listed in either of the following guides:
```
Comment (ameypv-rudder) [inline docs/sources/reverse-etl/features/syncing-events.mdx:36]:
```suggestion
2. Connect the source to the required destination by configuring the destination settings listed in either of the following guides:
```

File: docs/sources/reverse-etl/features/syncing-events.mdx [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+- <Link to="/sources/reverse-etl/common-settings/importing-data-using-tables/#data-import-settings">Importing Data using Tables</Link>: If you have configured your source using the Table option.
```
Comment (ameypv-rudder) [inline docs/sources/reverse-etl/features/syncing-events.mdx:37]:
```suggestion
- <Link to="/sources/reverse-etl/common-settings/importing-data-using-tables/#data-import-settings">Importing Data using Tables</Link>: If you have configured your source using the <strong>Table</strong> option.
```

File: docs/sources/reverse-etl/features/syncing-events.mdx [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+- <Link to="/sources/reverse-etl/common-settings/importing-data-using-models/#data-import-settings">Importing Data using Models</Link>: If you have configured your source using the Models option.
```
Comment (ameypv-rudder) [inline docs/sources/reverse-etl/features/syncing-events.mdx:38]:
```suggestion
- <Link to="/sources/reverse-etl/common-settings/importing-data-using-models/#data-import-settings">Importing Data using Models</Link>: If you have configured your source using the <strong>Models</strong> option.
```

File: docs/sources/reverse-etl/common-settings/importing-data-using-tables.mdx [hunk 11]
```diff
@@ -17,27 +17,42 @@ This guide lists the JSON mapping settings required to import and sync data from
...
 
 <div class="infoBlock">
-RudderStack supports two modes for syncing your data - <strong>Upsert</strong> and <strong>Mirror</strong>. For more information on these modes, refer to the <Link to="/sources/reverse-etl/common-settings/sync-modes/">Sync Modes</Link> guide.
+RudderStack uses the primary key column for diffing in case of incremental syncs. You can generate it using either of the following ways:
+<ul>
+<li>Generating your table with a primary key</li>
+OR
+<li>Creating a table view</li>
+</ul>
+The best combination for a primary key is the <code class="inline-code">timestamp</code> and <code class="inline-code">user_id</code>.
```
Comment (ameypv-rudder) [inline docs/sources/reverse-etl/common-settings/importing-data-using-tables.mdx:37]:
```suggestion
The best combination for a primary key is the event's <code class="inline-code">timestamp</code> and <code class="inline-code">userId</code>.
```

Comment (ameypv-rudder) [inline docs/sources/reverse-etl/common-settings/importing-data-using-tables.mdx:37]: @vyeshwanth quick question for my understanding - `timestamp` and `user_id` are the warehouse column names, right?

**Pull request merged**